source/adios2/helper/adiosCommMPI.cpp

```diff
@@ -189,6 +189,7 @@ CommImplMPI::~CommImplMPI()
             m_MPIComm != MPI_COMM_SELF)
         {
             MPI_Comm_free(&m_MPIComm);
+            m_MPIComm = MPI_COMM_NULL; // prevent freeing a second time
```

Comment:
The destructor shouldn't run more than once and no other methods of `CommImplMPI` should be called once the destructor finishes.  Therefore this is not necessary here.

The change in `CommImplMPI::Free` should be enough because only that method can reach the corresponding code path more than once.

Reply:
I'll test it to see if that's the case...